dev/com.ibm.ws.jca.cm/src/com/ibm/ejs/j2c/FreePool.java

```diff
@@ -248,9 +248,8 @@ protected void returnToFreePool(MCWrapper mcWrapper) {
         if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
             Tr.entry(this, tc, "returnToFreePool", gConfigProps.cfName);
         }
-        if (mcWrapper.shouldBeDestroyed() || mcWrapper.hasFatalErrorNotificationOccurred(fatalErrorNotificationTime)
-            || ((pm.agedTimeout != -1)
-                && (mcWrapper.hasAgedTimedOut(pm.agedTimeoutMillis)))) {
+        if (mcWrapper.shouldBeDestroyed() || mcWrapper.hasFatalErrorNotificationOccurred(fatalErrorNotificationTime) || !mcWrapper.isPoolable()
```

Comment:
This might be more a question for Jim, but should the method mcWrapper.shouldBeDestroyed() also be updated to consider the poolable=false value. If so, you would not need the `|| !mcWrapper.isPoolable()` here.

Reply:
The isPoolable()  is clear on its intended use.  

If we need to optimize this area of the code under a different pull, its likely we should have a discard boolean that is set for any connections that should be discarded, but still have additional support variables that represent why its being discarded for trace and debugging.   One discard boolean would clean up this code.

Reply:
@njr-11 I looked at other places we use `shouldBeDestroyed()` and it was typically used to check if their was already a connection error presented to the user.  Otherwise, these debug messages are hidden. For example:

https://github.com/OpenLiberty/open-liberty/blob/3a3e3a27c8ea077619740b911a7ce5530c0ad66d/dev/com.ibm.ws.jca.cm/src/com/ibm/ejs/j2c/XATransactionWrapper.java#L336-L339

Which is why I decided to add a new method